Test Failure: #Tile() #renderToVector() should render a GeoJSON

I am seeing a test failure:

```
  1) #Tile() #renderToVector() should render a GeoJSON:
     Uncaught AssertionError: {"type":"FeatureCollection","features":[{"type":"Feature","geometry":{"type":"Polygon","coordinates":[[[-71.71188354492186,19.71 == [123,34,116,121,112,101,34,58,34,70,101,97,116,117,114,101,67,111,108,108,101,99,116,105,111,110,34,44,34,102,101,97,116,117,114
      at /home/tom/rpm/BUILD/package/test/tile.js:53:24
```

There appear to be two problems - firstly the hash keys are not necessarily ordered the same when the first argument to the assert is formatted as in the example output. Using `deepEqual` should fix this I think.

The second is that some of the coordinates in the GeoJSON differ in the last decimal place, due to floating point comparison issues. There are ways to deal with this for simple comparisons of numeric values but I haven't yet found one that applies to numeric comparisons done as part of a deep comparison of a data structure...

--- a/test/tile.js
+++ b/test/tile.js
@@ -5,6 +5,20 @@ var Config = require('../src/Config.js').Config,
     assert = require('assert'),
     mapnik = require('mapnik');
 
+var trunc_6 = function(key, val) {
+    return val.toFixed ? Number(val.toFixed(6)) : val;
+}
+
+function compareGeoJSON(json1, json2) {
+    if (typeof json1 === 'string') json1 = JSON.parse(json1);
+    if (typeof json2 === 'string') json2 = JSON.parse(json2);
+
+    json1 = JSON.parse(JSON.stringify(json1, trunc_6));
+    json2 = JSON.parse(JSON.stringify(json2, trunc_6));
+
+    return assert.deepEqual(json1, json2);
+}
+
 describe('#Tile()', function () {
     var config, project, map;
 
@@ -50,7 +64,7 @@ describe('#Tile()', function () {
             var tile = new Tile(6, 19, 28);
             tile.renderToVector(project, map, function (err, vtile) {
                 if (err) throw err;
-                assert.deepEqual(JSON.parse(vtile.toGeoJSON('__all__')), JSON.parse(fs.readFileSync('test/data/expected/tile.world.6.19.28.geojson')));
+                compareGeoJSON(vtile.toGeoJSON('__all__'), JSON.parse(fs.readFileSync('test/data/expected/tile.world.6.19.28.geojson')));
                 done();
             });
         });

